parsel/selector.py

```diff
@@ -197,18 +200,19 @@ class Selector(object):
     selectorlist_cls = SelectorList
 
     def __init__(self, text=None, type=None, namespaces=None, root=None,
-                 base_url=None, _expr=None):
+                 base_url=None, _expr=None, encoding='utf8'):
         self.type = st = _st(type or self._default_type)
         self._parser = _ctgroup[st]['_parser']
         self._csstranslator = _ctgroup[st]['_csstranslator']
         self._tostring_method = _ctgroup[st]['_tostring_method']
 
         if text is not None:
-            if not isinstance(text, six.text_type):
-                msg = "text argument should be of type %s, got %s" % (
+            if isinstance(text, six.text_type) or isinstance(text, bytes):
+                root = self._get_root(text, base_url, encoding=encoding)
+            else:
+                msg = "text argument should be of type %s or bytes, got %s" % (
                     six.text_type, text.__class__)
                 raise TypeError(msg)
-            root = self._get_root(text, base_url)
```

Comment:
:lipstick: What about using `if not isinstance(text, (bytes, six.text_type)):` instead, to maintain the current early-raise approach that minimizes indentation?